### PR TITLE
WRKLDS-950: add since and since-time flags to limit log collection on must-gather

### PR DESCRIPTION
**Pull request by flavianmissi**
This needs to be tested with https://github.com/openshift/must-gather/pull/400.

### Build the must-gather image
1. Get the code from https://github.com/openshift/must-gather/pull/400
2. Build it:
```
QUAY_USER="fmissi"
podman build -t quay.io/${QUAY_USER}/must-gather -f Dockerfile.rhel7 .
podman push quay.io/${QUAY_USER}/must-gather
```

### Import the custom must-gather image into the cluster

```
QUAY_USER="fmissi"
oc project openshift
oc import-image my-must-gather --from=quay.io/${QUAY_USER}/must-gather --confirm
```

### Run oc must-gather
1. Get the code from this PR
2. Run the must-gather using the new `--since` OR `--since-time` options:
```
go run ./cmd/oc adm must-gather --since-time=2024-01-11T09:00:00Z --image-stream=openshift/my-must-gather:latest
go run ./cmd/oc adm must-gather --since=2h --image-stream=openshift/my-must-gather:latest
```

### CODE REVIEW

File: pkg/cli/admin/mustgather/mustgather.go [hunk 1]
```diff
@@ -795,6 +803,14 @@ func (o *MustGatherOptions) newPod(node, image string, hasMaster bool) *corev1.P
 					// always force disk flush to ensure that all data gathered is accessible in the copy container
 					Command: []string{"/bin/bash", "-c", fmt.Sprintf("%s & %s; sync", volumeUsageChecker, executedCommand)},
 					Env: []corev1.EnvVar{
+						{
+							Name:  "MUST_GATHER_SINCE",
```
Comment (ardaguclu) [inline pkg/cli/admin/mustgather/mustgather.go:807]: nit: I think, it is better to not set these environment variables, if since or since time are not set.

Comment (flavianmissi) [reply pkg/cli/admin/mustgather/mustgather.go]: done!

**Comment by ardaguclu** [comment]
Thanks

/lgtm
/hold
until prerequisite PR is merged and pre-merge tests are done and @ingvagabund also checks

**Comment by flavianmissi** [comment]
test failures seem unrelated
/retest

**Comment by flavianmissi** [comment]
test failure seem unrelated
/test e2e-aws-ovn-serial

**Comment by flavianmissi** [comment]
QE noticed that we did not enforce RFC3339 for the value of `--since-time`, so I added validation: https://github.com/openshift/oc/pull/1648/files#diff-8d6c382f82b8d8fbbbaa9fb012ce55635c2f60ae74d65f3e4eb8fe307f373dabR306-R310
@ardaguclu could you have another look?

**Comment by ingvagabund** [comment]
/hold cancel
LGTM

**Comment by flavianmissi** [comment]
don't we need to merge https://github.com/openshift/must-gather/pull/400 first?

**Comment by ardaguclu** [comment]
> don't we need to merge https://github.com/openshift/must-gather/pull/400 first?
> 

You are right. I planned to tag must-gather, instead tagged this one

**Comment by ardaguclu** [comment]
Actually we can merge this first, env variables will be ignored. Feel free to cancel hold

**Comment by flavianmissi** [comment]
let's wait - QE is still testing must-gather and I want to be sure everything works smooth :)

**Pull request merged**